server/plugin.go

```diff
@@ -206,3 +210,117 @@ func (p *Plugin) sendDirectMessage(userID string, message string) error {
 	_, err = p.API.CreatePost(post)
 	return err
 }
+
+func (p *Plugin) setUserStatus(userID string, meetingID int, meetingEnd bool) error {
+	storedStatusPref, appErr := p.API.KVGet(fmt.Sprintf("%v_%v", changeStatusKey, userID))
+	if appErr != nil {
+		p.API.LogDebug("Could not get stored status preference from KV ", appErr)
+	}
+
+	if storedStatusPref == nil {
+		err := p.sendStatusChangeAttachment(userID, p.botUserID, meetingID)
+		if err != nil {
+			p.API.LogDebug("could not send status change attachment ", "error", err)
+		}
+	}
+
+	if string(storedStatusPref) != yes {
+		return nil
+	}
+
+	statusKey := fmt.Sprintf("%v%v", oldStatusKey, meetingID)
+	if meetingEnd {
+		statusVal, err := p.API.KVGet(statusKey)
+		if err != nil {
+			p.API.LogDebug("Could not get old status from KVStore", "err", appErr.Error())
+			return err
+		}
+
+		newStatus := string(statusVal)
+		if newStatus == "" {
+			newStatus = model.STATUS_ONLINE
+		}
+
+		_, appErr = p.API.UpdateUserStatus(userID, newStatus)
+		if appErr != nil {
+			p.API.LogDebug("Could not get update status", "err", appErr.Error())
+			return appErr
+		}
+
+		return nil
+	}
+
+	currentStatus, appErr := p.API.GetUserStatus(userID)
+	if appErr != nil {
+		p.API.LogDebug("Failed to update user status", "err", appErr)
+		return appErr
+	}
+
+	oldStatus := ""
+	if currentStatus.Manual {
+		oldStatus = currentStatus.Status
+	}
+
+	appErr = p.API.KVSetWithExpiry(statusKey, []byte(oldStatus), meetingPostIDTTL)
+	if appErr != nil {
+		p.API.LogDebug("failed to store old status", "err", appErr)
+		return appErr
+	}
+
+	_, appErr = p.API.UpdateUserStatus(userID, model.STATUS_DND)
+	if appErr != nil {
+		p.API.LogDebug("Failed to update user status", "err", appErr)
+		return appErr
```

Comment:
Can we use `errors.Wrap` to decorate errors with relevant error messages in this function and have the parent log, instead of logging here?